docs/matchers/index.rst

```diff
@@ -0,0 +1,388 @@
+^^^^^^^^^^^^^^^^^^^^
+testslide.matchers.FloatGreaterOrEquals() represents an argument, that matches any Float, that is greater than or equals the specified one
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatGreaterOrEquals(6.9))\
+          .to_return_value(None)
+
+FloatLess
+^^^^^^^^^
+testslide.matchers.FloatLess() represents an argument, that is less than the specified Float
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatLess(6.9))\
+          .to_return_value(None)
+
+FloatLessOrEquals
+^^^^^^^^^^^^^^^^^
+testslide.matchers.FloatLessOrEquals() represents an argument, that is less than or equals the specified Float
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatLessOrEquals(6.9))\
+          .to_return_value(None)
+
+
+Strings
+-------
+
+AnyStr
+^^^^^^
+testslide.matchers.AnyStr() represents an argument, that matches any String
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.AnyStr())\
+          .to_return_value(None)
+
+RegexMatches
+^^^^^^^^^^^^
+testslide.matchers.RegexMatches() represents an argument, that is a String and matches provided regular expression.
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.RegexMatches(".*\.json"))\
+          .to_return_value(None)
+
+
+Collections
+-----------
+
+NotEmpty
+^^^^^^^^
+testslide.matchers.NotEmpty() represents an argument, that is a collection (eg: list, dict, tuple), and has elements.
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.NotEmpty())\
+          .to_return_value(None)
+
+Empty
+^^^^^
+testslide.matchers.NotEmpty() represents an argument, that is a collection (eg: list, dict, tuple), and has no elements.
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.Empty())\
+          .to_return_value(None)
+
+ListContaining
+^^^^^^^^^^^^^^
+testslide.matchers.NotEmpty() represents an argument, that is a list, and contains all the elements of the matcher.
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.ListContaining([1,2,3,"a"]))\
```

Comment:
What If I do `ListContaining(3)`? I can see people getting confused by it, "does it contains each element or a single element which's a list?". Also, "in which order?". Personally I'd go this being true if "a in list" is true.